akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -499,12 +495,17 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
       byteBuffer.putShort(length.toShort)
       ensureLiteralCharsLength(length)
       val bytes = literalBytes
-      val chars = Unsafe.instance.getObject(literal, StringValueFieldOffset).asInstanceOf[Array[Char]]
-      var i = 0
-      while (i < length) {
-        // UsAscii
-        bytes(i) = chars(i).asInstanceOf[Byte]
-        i += 1
```

Comment:
@ktoso @patriknw a hack to compensate another hack, not pretty but should get the job done without losing performance, in fact; in Java 9 because of `System.arraycopy` and no casting back `byte` to `char` it should be slightly faster.

Reply:
I'd propose to encapsulate the difference in our Unsafe, and we could use mr-jars to hide the version switching from code so we don't have to think about it when having to do such thing.

Reply:
I could do that, build stuff could be a bit confusing to set up at first, but shuold be a good thing to do in any case IMHO

Reply:
@ktoso looks better now IMHO, also added benchmark to point to the new place where Unsafe copy is happening.